contracts/id.tmy/src/id.tmy.cpp

```diff
@@ -178,6 +197,24 @@ namespace idtmy
                            app_itr.origin = origin;
                            app_itr.username_hash = username_hash;
                            app_itr.version = 1; });
+
+      // Store the account type in the account_types table
+      account_types_table account_types(get_self(), get_self().value);
+      account_types.emplace(get_self(), [&](auto& row) {
+         row.account_name = random_name;
+         row.account_type = enum_account_type::App;
+      });
+      
+      eosiobios::bios::resource_config_table _resource_config("eosio.bios"_n, "eosio.bios"_n.value);
+      int64_t myRAM, net, cpu;
+      eosio::get_resource_limits( app_name, myRAM, net, cpu );
+      _resource_config.emplace(get_self(), [&](auto &row) {
+         row.total_cpu_weight_allocated = this->initial_cpu_weight_allocation;
+         row.total_net_weight_allocated = this->initial_net_weight_allocation;
+      });
+
+      eosio::set_resource_limits(app_name, myRAM, this->initial_cpu_weight_allocation, this->initial_net_weight_allocation);
```

Comment:
do this earlier. actions should have structure

1. check/validate arguments
2. check conditions
3. use CDT API e.g. resource allocation
4. modify state
5. call inline actions